File: benchmark/2026-09-17-7677581(latest)/semantic-compression/probes/java/F14.P3.java
import shapes.Shape;

class Main {
    static double triArea() {
        // BEGIN PROBE F14.P3
        record Tri(double b, double h) implements Shape {
            public double area() {
                return 0.5 * b * h;
            }
        }
        Shape s = new Tri(3.0, 4.0);
        return s.area();
        // END PROBE F14.P3
    }

    public static void main(String[] args) {
        System.out.println(triArea());
    }
}

File: benchmark/2026-09-17-7677581(latest)/semantic-compression/probes/zig/F14.P3.zig
const std = @import("std");
const shapes = @import("shapes.zig");

fn tri_area() f64 {
// BEGIN PROBE F14.P3
const Tri = struct {
    b: f64,
    h: f64,
    fn area(p: *const anyopaque) f64 {
        const self: *const @This() = @ptrCast(@alignCast(p));
        return 0.5 * self.b * self.h;
    }
};
const t: Tri = .{ .b = 3.0, .h = 4.0 };
const s: shapes.Shape = .{ .ptr = &t, .areaFn = Tri.area };
return s.area();
// END PROBE F14.P3
}

pub fn main() !void {
    var threaded: std.Io.Threaded = .init_single_threaded;
    const io = threaded.io();
    var out_buf: [64]u8 = undefined;
    var out = std.Io.File.stdout().writer(io, &out_buf);
    try out.interface.print("{d}\n", .{tri_area()});
    try out.interface.flush();
}
